packages/@ember/-internals/glimmer/lib/helpers/and.ts

```diff
@@ -0,0 +1,37 @@
+import { Arguments, CapturedArguments, VM } from '@glimmer/runtime';
+import { InternalHelperReference } from '../utils/references';
+
+/**
+@module ember
+*/
+
+/**
+  Evaluates the given arguments with && in short-circuit
+
+  Example:
+
+  ```handlebars
+  {{and isAdmin isIdle validData}}
+
+  {{! be validData if `isAdmin` and `isIdle` are both truthy}}
+  ```
+
+  @public
+  @method and
+  @for Ember.Templates.helpers
+  @since 2.7.0
+*/
+function and({ positional: { references } }: CapturedArguments) {
+  let last: any = true;
+  for (let i = 0; i < references.length; i++) {
+    last = references[i].value();
+    if (!last) {
```

Comment:
@chancancode - Should we consider `[]` as falsey here? In general, in template land I think we do right?

Reply:
yeah, I think we do. https://github.com/emberjs/ember.js/blob/1b7e613f4455c929010b59dd8d65c0b892ca3b08/packages/%40ember/-internals/glimmer/lib/utils/to-bool.ts#L4-L10

I bigger question is should we support proxies, I completely forgot about that. I think that needed to be discussed and clarified in the RFC 🤧

Reply:
That is surprising. I was expecting and/or helpers to behave just like && / II. Why this deviation from standard JS behaviour?

Reply:
```hbs
{{#each @posts as |post|}}
  ...
{{else}}
  No posts!
{{/each}}
```

Reply:
Well it really isn't about `each`, `{{#if someEmptyArray}}{{else}}{{/if}}` will render the falsey branch. I think it is very confusing to have `if` behave differently than `eq`/`not`.

Reply:
It works the same way for...

```hbs
{{#if @posts as |post|}}
  ...
{{else}}
  No posts!
{{/if}}
```

...and `with`, `let` etc. I think the bottom line is it would be pretty surprising if `{{else}}` doesn't work consistently in the control flow things in the "language".

So now what if you refactor from the first example into...

```hbs
{{#each (or @posts @reblogs) as |post|}}
  ...
{{else}}
  No posts!
{{/each}}
```

...what should this do?

Reply:
I think it is probably important for the "language" to have one boolean semantics, and for better or for worse empty arrays are falsey everywhere else.